pkg/controller/deployment/deployment_controller.go

```diff
@@ -581,6 +581,20 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	// This is the point at which we used to add/remove the overlap annotation.
```

Comment:
Add to this comment a date at which this code can be removed + file an issue and target a milestone.